File: sample/src/main/java/com/like/location/sample/MainActivity.java
package com.like.location.sample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.baidu.location.BDLocation;
import com.baidu.trace.api.entity.EntityListResponse;
import com.baidu.trace.api.entity.OnEntityListener;
import com.baidu.trace.api.track.HistoryTrackResponse;
import com.baidu.trace.api.track.OnTrackListener;
import com.like.location.LocationUtils;
import com.like.location.MyLocationListener;
import com.like.location.TraceUtils;
import com.like.logger.Logger;

public class MainActivity extends AppCompatActivity {
    private LocationUtils mLocationUtils;
    private TraceUtils mTraceUtils;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initLocation();
        mTraceUtils = new TraceUtils(this, 164746, 5, 10);
        mTraceUtils.startTrace();
    }

    /**
     * 初始化百度定位
     */
    public void initLocation() {
        mLocationUtils = new LocationUtils(this, new MyLocationListener() {
            @Override
            public void onReceiveLocation(BDLocation location) {
                super.onReceiveLocation(location);
                if (location != null) {
                    Logger.i("Location", location.getCity());
                }
                mLocationUtils.stop();
            }
        });
    }

    public void startLocation(View view) {
        mLocationUtils.start();
    }

    public void stopLocation(View view) {
        mLocationUtils.stop();
    }

    public void startTrack(View view) {
        mTraceUtils.startTrace();
    }

    public void stopTrack(View view) {
        mTraceUtils.stopTrace();
    }

    public void startGather(View view) {
        mTraceUtils.startGather();
    }

    public void stopGather(View view) {
        mTraceUtils.stopGather();
    }

    public void queryEntityList(View view) {
        mTraceUtils.queryEntityList(1, null, new OnEntityListener() {
            @Override
            public void onEntityListCallback(EntityListResponse entityListResponse) {
                Logger.e(entityListResponse);
            }
        });
    }

    public void queryHistoryTrack(View view) {
        mTraceUtils.queryHistoryTrack(1, System.currentTimeMillis() / 1000 - 12 * 60 * 60, System.currentTimeMillis() / 1000, new OnTrackListener() {
            @Override
            public void onHistoryTrackCallback(HistoryTrackResponse historyTrackResponse) {
                Logger.e(historyTrackResponse);
            }
        });
    }
}

File: sample/src/main/java/com/like/location/sample/MainActivity.kt
package com.like.location.sample

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.baidu.location.LocationClientOption
import com.like.location.LocationUtils
import com.like.location.NavigationUtils
import com.like.location.sample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val mBinding: ActivityMainBinding by lazy {
        DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
    }
    private val mLocationUtils: LocationUtils by lazy {
        LocationUtils()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding
        mLocationUtils.init(this)
    }

    fun getLocation(view: View) {
        val locationOption = LocationClientOption()
        // 设置定位场景，根据定位场景快速生成对应的定位参数  以出行场景为例
        // 1）签到场景：只进行一次定位返回最接近真实位置的定位结果
        // 2）运动场景：高精度连续定位，适用于运动类开发者场景
        // 3）出行场景：高精度连续定位，适用于运动类开发者场景
        locationOption.setLocationPurpose(LocationClientOption.BDLocationPurpose.SignIn)
        mLocationUtils.setLocationClientOption(locationOption)
        mLocationUtils.start()
    }

    fun start(view: View) {
        mLocationUtils.setLocationClientOption(mLocationUtils.getDefaultLocationClientOption())
        mLocationUtils.start()
    }

    fun stop(view: View) {
        mLocationUtils.stop()
    }

    fun reStart(view: View) {
        mLocationUtils.restart()
    }

    fun navigation(view: View) {
        NavigationUtils.navigation(this, 29.0, 106.0)
    }

    fun map(view: View) {
        startActivity(Intent(this, MapActivity::class.java))
    }

    override fun onDestroy() {
        super.onDestroy()
        mLocationUtils.onDestroy()
    }
}
